lib/ansible/modules/system/osx_defaults.py

```diff
@@ -15,413 +16,975 @@
 DOCUMENTATION = '''
 ---
 module: osx_defaults
-author: Franck Nijhof (@frenck)
-short_description: osx_defaults allows users to read, write, and delete Mac OS X user defaults from Ansible
+author: "Franck Nijhof (@frenck), Etienne Desautels (@etienned)"
+short_description:
+    Allows users to read, write, and delete Mac OS X preferences from Ansible.
 description:
-  - osx_defaults allows users to read, write, and delete Mac OS X user defaults from Ansible scripts.
-    Mac OS X applications and other programs use the defaults system to record user preferences and other
-    information that must be maintained when the applications aren't running (such as default font for new
-    documents, or the position of an Info panel).
+  - osx_defaults allows users to read, write, and delete Mac OS X preferences
+    from Ansible scripts.
+    Mac OS X applications and other programs use the defaults system to record
+    user preferences and other information that must be maintained when the
+    applications aren't running (such as default font for new documents, or
+    the position of an Info panel).
 version_added: "2.0"
+requirements:
+  - Python 2.6+.
+  - OS X installed Foundation and CoreFoundation python modules (or from PyObjC).
 options:
   domain:
     description:
-      - The domain is a domain name of the form com.companyname.appname.
-    required: false
+      - The I(domain) is a domain name of the form com.companyname.appname or
+        I(NSGlobalDomain) for the Global Domain.
     default: NSGlobalDomain
   host:
     description:
       - The host on which the preference should apply. The special value "currentHost" corresponds to the
         "-currentHost" switch of the defaults commandline tool.
-    required: false
     default: null
     version_added: "2.1"
+  any_user:
```

Comment:
I would suggest renaming this to `user` with choices `["currentUser", "anyUser"]` and `default: currentUser` for better consistency with host.

Reply:
As I don't think I'll make your previous suggestion about `host`, consistency is not an issue here. Also, as they are only two values possible, it's simpler for the users to treat it as a boolean, so users don't have to remember what are the choices.